### PR TITLE
Some improvements

### LINKED ISSUE #1045: Add notify_admin_gui to check_command
Also remove `occ_command` from `notify_admin_gui`

### PR DESCRIPTION
**Pull request by szaimen**
Fix #1045
- [x] Add notify_admin_gui to check_command
- [x] remove occ_command from notify_admin_gui
- [x] reenable the app updates admin notifications
- [x] create occ_command_no_check function
- [x] revert changes to the updatenotification app
- [x] improve notify_admin_gui based on kyrofas implementation

### CODE REVIEW

**Comment by szaimen** [comment]
@enoch85 this should be ready for review.

File: nextcloud_update.sh [hunk 1]
```diff
@@ -309,25 +309,22 @@ fi
...
 then
-    # Make sure maintenance:mode isn't activated (it will fail if it is)
-    occ_command maintenance:mode --off
+    occ_command_no_check maintenance:mode --off
     # Check for upgrades
     print_text_in_color "$ICyan" "Trying to automatically update all Nextcloud apps..."
-    # occ command can not be used due to the check_command() function.
-    sudo -u www-data php $NCPATH/occ app:update --all
-#    UPDATED_APPS="$(sudo -u www-data php $NCPATH/occ app:update --all)"
+    UPDATED_APPS="$(occ_command_no_check app:update --all)"
```
Comment (szaimen) [inline nextcloud_update.sh]: Can't we use here the normal occ_command?

**Comment by szaimen** [comment]
BTW: I am not sure if  `Add notify_admin_gui to check_command` really makes sense

**Comment by enoch85** [comment]
> > Add notify_admin_gui to check_command
> 
> BTW: I am not sure if that really makes sense

If there are several admins, they might want to know whats going on, also we would get more issue reports to fix bugs, which is a good thing.

**Comment by enoch85** [comment]
- [ ] Check if sha256sum is installed before checking sum of the downloaded Nextcloud  tar.

File: nextcloud_update.sh (hunk 1 above)
Comment (enoch85) [reply nextcloud_update.sh]: > occ command can not be used due to the check_command() function.

No :)

**Comment by szaimen** [comment]
We could add a check to notify_admin_gui to check whether the notification app is enabled.
and btw: this is how kyrofa has implemented the admin notifications:
https://github.com/nextcloud/nextcloud-snap/pull/1221/files#diff-d98160f676559fde3039f591da87ca8f

**Comment by szaimen** [comment]
We could add a check to notify

**Comment by szaimen** [comment]
We could add a check to notify

**Comment by szaimen** [comment]
Sry, pressed the wrong button

**Comment by szaimen** [comment]
Actually, I would vote to revert the changes made to the upgrade notification app:
it is stilll quite useful: it shows you which apps aren't compatible with the next version and you can see, if the next major release is already stable:
which isn't the case in the moment, although it is released as stable.
(if you check for updates in the stable channel, it doesn't show you the 18.0 release, only if you switch to beta)

**Comment by enoch85** [comment]
> Actually, I would vote to revert the changes made to the upgrade notification app:
it is stilll quite useful: it shows you which apps aren't compatible with the next major version (and need to be updated) and you can see, if the next major release is already stable:
which isn't the case in the moment with 18.0, although it is released as stable.
(if you check for updates in the stable channel, it doesn't show you the 18.0 release, only if you switch to beta)

Please do what you think is best. I noticed today that I got info about new app updates from the notification app anyway, so maybe adding it in the VM is a bit overkill?

I need to sort the tons of emails I have before I can look at this. Sorry.

**Comment by szaimen** [comment]
> Please do what you think is best. I noticed today that I got info about new app updates from the notification app anyway, so maybe adding it in the VM is a bit overkill?
> 
> I need to sort the tons of emails I have before I can look at this. Sorry.

I will do that later on.
The app updates are from the upgrade-notification app. If it is disabled, you don't get anymore.
I don't think, that it is overkill. If you don't install the app updates manually, they get automatically installed by the update script, if enabled. I think it is good to know, if they got installed.

**Comment by szaimen** [comment]
> > Please do what you think is best. I noticed today that I got info about new app updates from the notification app anyway, so maybe adding it in the VM is a bit overkill?
> > I need to sort the tons of emails I have before I can look at this. Sorry.
> 
> I will do that later on.
> The app updates notifications are from the upgrade-notification app. If it is disabled, you don't get them anymore.
> I don't think, that informing the user via script it is overkill. If you don't install the app updates manually, they get automatically installed by the update script, if enabled. I think it is good to know, if they got installed.

the changes to updatenotification are reverted

**Comment by szaimen** [comment]
> We could add a check to notify_admin_gui to check whether the notification app is enabled.
> and btw: this is how kyrofa has implemented the admin notifications in the snap:
> https://github.com/nextcloud/nextcloud-snap/pull/1221/files#diff-d98160f676559fde3039f591da87ca8f
> I assume he has taken the output format json, because he thinks that this output format will be more stable in the future.

What do you think about that?

**Comment by enoch85** [comment]
> What do you think about that?

@kyrofa's implementation seems good to me, please improve our.

**Comment by szaimen** [comment]
> > What do you think about that?
> 
> @kyrofa's implementation seems good to me, please improve our.

I will do that later on. First have to test, if it works in the VM.

**Comment by szaimen** [comment]
> > > What do you think about that?
> > 
> > 
> > @kyrofa's implementation seems good to me, please improve our.
> 
> I will do that later on. First have to test, if it works in the VM.

This should be done.

**Comment by szaimen** [comment]
@enoch85 what about your checkpoints here: https://github.com/nextcloud/vm/pull/1046#issuecomment-576055408

**Pull request merged**